modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -219,6 +220,22 @@ public void addObjectRelationshipMappingTableValues(
 			return;
 		}
 
+		ObjectEntry objectEntry1 = _objectEntryLocalService.getObjectEntry(
+			primaryKey1);
+
+		ObjectEntry objectEntry2 = _objectEntryLocalService.getObjectEntry(
+			primaryKey2);
```

Comment:
You're not considering that the `primaryKey` might belong to an **unmodifiable system object definition**.

Note that at the beginning of this method, we already have a call to `_validateObjectEntryId(objectDefinition1, primaryKey1);`

I think we can reuse this method. Please:
Make it return the ObjectEntry (and rename it accordingly).

Move your logic to before the M-M / 1-M conditions.